# Homework 4 Assignment 3

In [29]:
from __future__ import print_function
import os
import numpy as np
import pandas as pd
%pylab inline

Populating the interactive namespace from numpy and matplotlib


**Null hypothesis**: The mean commute time at peak hours is at 36 minutes or longer with the new bus route X8 as it was before, with a _significance level of p=0.05_


**Alternative hypothesis**: The mean commute time at peak hours is shorter than 36 minutes with the new bus route X8 than it was before. 

_Significance level of p=0.05_

$H_0$: $p_1$ - $p_0$ => 0

$H_1$: $p_1$ - $p_0$ < 0

α = 0.05

where if z-value is 2 or more signmas away from old mean, we will reject  null hypothesis

In [30]:
!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt > x8bus.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0   5457      0 --:--:-- --:--:-- --:--:--  5483


In [31]:
os.environ["PUIDATA"] = "%s/001_PUI/PUIDATA"%os.getenv("HOME")
print("Path to PUIDATA", os.getenv("PUIDATA"))

Path to PUIDATA /nfshome/qg412/001_PUI/PUIDATA


In [32]:
!mv x8bus.txt $PUIDATA

In [39]:
#reading the text file as a table
x8_df = pd.read_table("%s/x8bus.txt"%os.getenv("PUIDATA"), delim_whitespace = True, names=('t'))
x8_df.head(5)

t
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

In [42]:
# calculate the sample mean
samplemean = x8_df.mean()
samplemean

t    34.466162
dtype: float64

The observed mean is smaller than the population mean. so now we need to check if it is statistically significantly lower than the population mean.

In [44]:
# calculate the sample standard deviation
sample_sd = x8_df.std()
sample_sd

t    7.13728
dtype: float64

In [47]:
# calculate the sample size
samplesize = len(x8_df)
samplesize

100

In [48]:
popmean = 36
popsd = 6

In [50]:
# calculate z statistics

ztest = (popmean - samplemean)/(popsd/(samplesize**0.5))
ztest

t    2.556397
dtype: float64

## Analysis:

The z-statistics of 2.556 is more than 2 sigmas away from the old mean therefore we reject the null hypothesis that new bus line